main.tf

```diff
@@ -355,3 +355,58 @@ resource "ibm_is_flow_log" "flow_logs" {
 }
 
 ##############################################################################
+# DNS ZONE
+# ##############################################################################
+
+resource "ibm_dns_zone" "dns_zone" {
+  count       = var.enable_hub && !var.skip_custom_resolver_hub_creation ? 1 : 0
+  name        = var.dns_zone_name
+  instance_id = var.use_existing_dns_instance ? var.existing_dns_instance_id : ibm_resource_instance.dns_instance_hub[0].guid
+  description = var.dns_zone_description
+  label       = var.dns_zone_label
+}
+
+##############################################################################
+# DNS PERMITTED NETWORK
+##############################################################################
+
+resource "ibm_dns_permitted_network" "dns_permitted_network" {
+  count       = var.enable_hub && !var.skip_custom_resolver_hub_creation ? 1 : 0
+  instance_id = var.use_existing_dns_instance ? var.existing_dns_instance_id : ibm_resource_instance.dns_instance_hub[0].guid
+  zone_id     = ibm_dns_zone.dns_zone[0].zone_id
+  vpc_crn     = local.vpc_crn
+  type        = "vpc"
+}
+
+##############################################################################
+# DNS Records
+##############################################################################
+
+resource "ibm_dns_resource_record" "dns_record" {
+
+  for_each    = { for idx, record in var.dns_records : idx => record }
+  instance_id = var.use_existing_dns_instance ? var.existing_dns_instance_id : ibm_resource_instance.dns_instance_hub[0].guid
+  zone_id     = ibm_dns_zone.dns_zone[0].zone_id
```

Comment:
One thing I realised is that `ibm_dns_zone` is created conditionally, and this resource does not. It would be better to see if there's a value for `ibm_dns_zone.dns_zone` (count > 0), then only use it. This might fail otherwise with out-of-bound error.

Reply:
Thank you for mentioning this. I have made this change.